### ISSUE TITLE
Proposal for grouping of changelog

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#8596]
Currently we group the changelog entries mostly by change / new / fix, which are internal categories from a (mostly) technical point of view. I propose to introduce a grouping by functionality to make it easier for our users to understand whether they might be affected by a change or not. Functional groups could be something like "map", "connector", etc.

Based on today's changelog_master I tried to add the proposed grouping:

```
Map:
· New: Offer a link to download OpenStreetMap offline maps\n
· New: Integrate downloaded OpenStreetMap offline map files into c:geo after tapping on them\n
· New: Optionally switch between compact and regular icon modes automatically depending on amount of caches displayed\n
· Change: Renamed trail / trail history / history trail to history track\n
· New: History track: Allow export as GPX track file\n
· New: History track: Make max. length configurable\n
· New: Load GPX track files and show individual track on map\n
· New: Individual routes: Allow sorting of route elements\n
· New: Customize history track, circle, individual route and individual track map line colors and widths\n

Cache list:
· Change: Difficulty and terrain filter match exact values only, no longer ranges\n
· Change: Fast scroll only gets enabled after fling gesture with minimum length\n

Cache details:
· Change: Open links in cache description as separate task\n
· Fix: Adapt Whereigo URL checker to catch more Whereigo links\n

Supported geocaching platforms:
· Change: Set all connectors to https\n
· New: Support cache rating and managing of favorites for geocaching.su\n
· Fix: Adapt to updated terracaching details URL\n

System requirements:
· Change: System requirements: c:geo now requires your system to run at least Android 5.0\n
· Change: Set target SDK to 29 to comply with upcoming requirements for Play store\n
· New: Internal: Upgraded several included libraries to reflect our new minimum system requirements\n
· Fix: Internal: Removed styles / themes workarounds used for older Android versions\n
· Fix: Internal: Remove old program code no longer used for supported Android versions\n

Other changes:
· Change: New adaptive launcher icon, following the new specs from Play store (also used on mainscreen)\n
· New: Add debug view for program settings\n
· Fix: Fix for slider preference problem on Android 7\n
· Fix: Fix translation issues for several plural forms\n
· Fix: Allow restore of settings backup created with c:geo version 2020.03.30 or earlier\n
· Fix: Other code cleanup\n
```

What do you think?

**Comment by Lineflyer** [comment]
Fine with that, as long as we agree in the categories (fixed) and in a way not to many changes end up in "Other"

**Comment by Lomanic** [comment]
Not sure I should create an issue for this but now that the changelog is being discussed, I would appreciate a link back to the fixed Github issue for each line of the changelog.

**Comment by Lineflyer** [comment]
To much manual effort and probably not meaningful or helpful for the standard user.
You can always easily look up all changes to the code on Github if you are interested to do so.

**Comment by moving-bits** [comment]
@Lomanic
https://github.com/cgeo/cgeo/commits/release would give you the commit history for the release version. If a commit string contains the phrase "fix #xxxx" or "fixes #xxxx" it's a fix for the mentioned GitHub issue.

**Comment by moving-bits** [comment]
> Fine with that, 

Ok, implemented current proposal. Let's see how it works.

> as long as we agree in the categories (fixed) 

Categories will evolve over time, IMHO. Not sure, if we need to define a standard set. It probably depends of the focus of a version and how those changes can be grouped into a meaningful context.

> and in a way not to many changes end up in "Other"

I do understand your concern and generally agree, but until today everything was "category other" ;-)

In my proposal I tried to find groups which have a meaning for the "average user", and which have at least two entries in the changelog. Everything else went to "other" for now. But that might change with upcoming commits, if a new group satisfying above rule evolves with the new commit.

**Comment by moving-bits** [comment]
Released with 2020.08.04 version, can close here.
